src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -0,0 +1,157 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeQuality.Analyzers.Maintainability
+{
+    /// <summary>
+    /// 
+    /// 
+    /// </summary>
+    public abstract class UseNameofInPlaceOfStringAnalyzer<TSyntaxKind> : DiagnosticAnalyzer
+        where TSyntaxKind : struct
+    {
+        // TODO: need a RuleId
+        internal const string RuleId = "NAMEOFANALYZER";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringTitle), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringMessage), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringDescription), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleWithSuggestion = new DiagnosticDescriptor(RuleId,
+                                                                         s_localizableTitle,
+                                                                         s_localizableMessage,
+                                                                         DiagnosticCategory.Maintainability,
+                                                                         DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                         isEnabledByDefault: true,
+                                                                         description: "Use nameof",
+                                                                         // TODO: add MSDN url
+                                                                         helpLinkUri: "http://msdn.microsoft.com/library/ms182181.aspx",
+                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleWithSuggestion);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            // TODO correct setting?
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterSyntaxNodeAction(AnalyzeArgument, ArgumentSyntaxKind);
+        }
+
+        protected abstract TSyntaxKind ArgumentSyntaxKind { get;  }
+
+        private void AnalyzeArgument(SyntaxNodeAnalysisContext context)
```

Comment:
Sorry for not syncing with you earlier on this, but the implementation will be a lot simpler if implemented as an IOperation analyzer. I will talk with you offline.